design-document.md

```diff
@@ -232,6 +232,352 @@ Note: Two non-colluding aggregators --- one leader and one standard aggregator
 --- are required to provide protect the inputs' privacy. Additional aggregators
 may be used to make the system more resilient; see {{CITE}}.
 
+## Ingestion {#proto-ingestion}
+
+[[TODO: Decide how to authenticate the leader in leader-to-helper connections.
+One option is to use client certificates for TLS; another is to have the leader
+sign its messages directly, as in Prio v2.]]
+
+Data ingestion involves the client uploading its secret shared inputs and the
+aggregators collectively validating the corresponding input. Before this process
+begins, the following must be true (the acronym "PA" stands for "Private
+Aggregation"):
+1. The client knows the URL of the leader endpoint, e.g., `example.com/metrics`.
+   We write this URL as `[leader]` below. (We write `[helper]` for the helper's
+   URL and `[aggregator]` for the URL of some aggregator.)
+1. The client has a root certificate it can use to authenticate the leader.
+1. The leader's root certificates are sufficient to authenticate each helper.
+1. The helper has chosen an HPKE key pair.
+1. The aggregators agree on the PA parameters associated with a given PA task.
+
+The PA parameters are structured as follows:
+
+```
+struct {
+  PATask task;
```

Comment:
Is there a 1:1 relationship between `PAParam` and `PATask`? I think yes, since participants need to be able to use a task ID to look up a `PAParam` value.

Reply:
Not necessarily. In any case, `PATask` is supposed to encode what's common between `PAParam` and `PAClientParam`.

Reply:
OH wait, I misunderstood. Yes, there's 1:1 map.